.github/workflows/lint.yml

```diff
@@ -13,28 +13,14 @@ env:
   FORCE_COLOR: "1"
 
 jobs:
-  ruff:
+  pre-commits:
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Python
-      uses: actions/setup-python@v4
-      with:
-        python-version: 3
-    - name: Install pip
-      run: python -m pip install --upgrade pip
-
-    - name: Install known good Ruff
-      run: python -m pip install ruff==0.0.261
-    - name: Lint with known good Ruff
-      run: ruff . --diff --format github
-
-    - name: Install latest Ruff
-      run: python -m pip install --upgrade ruff
-    - name: Lint with Ruff
-      continue-on-error: true
-      run: ruff . --diff --format github
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
```

Comment:
I'm following the guidelines of the github actions: https://github.com/actions/setup-pytho

> The python-version input is optional. If not supplied, the action will try to resolve the version from the default .python-version file. If the .python-version file doesn't exist Python or PyPy version from the PATH will be used. The default version of Python or PyPy in PATH varies between runners and can be changed unexpectedly so we recommend always setting Python version explicitly using the python-version or python-version-file inputs.

3.10 seems like a reasonable choice to avoid bad surprises. Anything you would me to change ?

Reply:
Well, prior to this, we used `python-version: 3`, and I think using the same thing as before is good. Also, we used python because we wanted to install ruff using pip. But do we still need it ? You are not running any Python command in the pre-commit step so..

Reply:
you need python to run the pre-commit action : https://github.com/pre-commit/action.
i I only set it to python 3 it will not be consistent on all runners (from what I understand) so I'll let you decide but I personnaly thin a fixed python version is safer

Reply:
With `python-version: 3`, runners use the latest cached version, usually the latest released version from Python (currently, Python 3.11). You won't be consistent *only* if you are omit the `python-version` field. But, pinning it to "latest major" (i.e. "3") is fine.